src/main/scala/chisel3/stage/ChiselAnnotations.scala

```diff
@@ -145,15 +145,92 @@ object ChiselGeneratorAnnotation extends HasShellOptions {
     ChiselGeneratorAnnotation(gen)
   }
 
+  /** Construct a [[ChiselGeneratorAnnotation]] with a generator function that
+    * will try to construct a Chisel Module from using that Module's name and a
+    * single parameters object. The Module must both exist in the class path and
```

Comment:
can you clarify the commends, so a parameters object is something that is like a case class that is trivially deserialized from JSON? Or it is a class name that points to a specific Object?